lms/djangoapps/bookmarks/views.py

```diff
@@ -0,0 +1,301 @@
+
+        Create/Post a new bookmark for particular Xblock.
+
+    **Example Requests**
+
+          GET /api/bookmarks/v1/bookmarks/?course_id={course_id1}
+
+          POST /api/bookmarks/v1/bookmarks/
+
+    **Response Values**
+
+        * count: The number of bookmarks in a course.
+
+        * next: The URI to the next page of bookmarks.
+
+        * previous: The URI to the previous page of bookmarks.
+
+        * num_pages: The number of pages listing bookmarks.
+
+        * results:  A list of bookmarks returned. Each collection in the list
+          contains these fields.
+
+            * id: String. The identifier string for the bookmark: {user_id},{usage_id}.
+
+            * course_id: String. The identifier string of the bookmark's course.
+
+            * usage_id: String. The identifier string of the bookmark's XBlock.
+
+            * display_name: (optional) String. Display name of the XBlock.
+
+            * path: (optional) List. List of dicts containing {"usage_id": "", display_name:""} for the XBlocks
+                from the top of the course tree till the parent of the bookmarked XBlock.
+
+            * created: ISO 8601 String. The timestamp of bookmark's creation.
+
+    """
+    authentication_classes = (OAuth2Authentication, SessionAuthentication)
+    permission_classes = (permissions.IsAuthenticated,)
+
+    paginate_by = 30
+    max_paginate_by = 500
+    paginate_by_param = 'page_size'
+    pagination_serializer_class = PaginationSerializer
+    serializer_class = BookmarkSerializer
+
+    def get_serializer_context(self):
+        """
+        Return the context for the serializer.
+        """
+        context = super(BookmarksView, self).get_serializer_context()
+        if self.request.method == 'POST':
+            return context
+        optional_fields_param = self.request.QUERY_PARAMS.get('fields', [])
+        optional_fields = optional_fields_param.split(',') if optional_fields_param else []
+        context['fields'] = DEFAULT_FIELDS + [field for field in optional_fields if field in OPTIONAL_FIELDS]
+        return context
+
+    def get_queryset(self):
+        course_id = self.request.QUERY_PARAMS.get('course_id', None)
+
+        bookmarks_queryset = Bookmark.objects.filter(user=self.request.user)
+
+        if not course_id:
+            return bookmarks_queryset.order_by('-created')
+        try:
+            course_key = CourseKey.from_string(course_id)
+        except InvalidKeyError:
+            log.error("Invalid course id '{course_id}'")
+            return []
+
+        return bookmarks_queryset.filter(course_key=course_key).order_by('-created')
+
+    def post(self, request):
+        """
+        POST /api/bookmarks/v1/bookmarks/
+        """
+        if not request.DATA:
+            error_message = _("No data provided")
+            return Response(
+                {
+                    "developer_message": error_message,
```

Comment:
Developer messages should not be translated. If you want to use the same string, you can use ugettext_noop like i /edx-platform/openedx/core/djangoapps/user_api/preferences/api.py